operations/helm/charts/mimir-distributed/Chart.yaml

```diff
@@ -1,8 +1,8 @@
 apiVersion: v2
-version: 4.1.0-weekly.225
-appVersion: r225
+version: 4.2.0
```

Comment:
Let's release from a release branch i.e. mimir-distributed-release-4.2 and not main like for mimir. The release branch is used to keep the documentation in sync  with the release.

We need to improve the https://github.com/grafana/mimir/blob/main/operations/helm/charts/mimir-distributed/RELEASE.md doc a bit, it doesn't mention yet the release branch for helm charts. Which is needed to be able to push the documentation.

Reply:
Also usually we do a release candidate first.

Reply:
> Let's release from a release branch i.e. mimir-distributed-release-4.2 and not main. 

OK, I've updated the PR to target `mimir-distributed-release-4.2` instead of `main`. I'll (separately) update the release docs to make this clearer as well.

> Also usually we do a release candidate first.

I think in this case we're safe to skip a release candidate - the only changes from the previous release are the GEM image bump and #4230.

Reply:
You're right, verified with 
```
git diff origin/mimir-distributed-release-4.1..origin/charleskorn/helm-4.2-release operations/helm/charts/mimir-distributed/
```